.ci/Jenkinsfile

```diff
@@ -170,3 +196,79 @@ def pushDockerImage(){
+          // Check if the package revision is published
+          def responseCode = httpRequest(method: "HEAD",
+            url: "https://package-storage.elastic.co/artifacts/packages/${packageName}-${packageVersion}.zip",
+            response_code_only: true)
+          if (responseCode == 200) {
+            continue // already published
+          }
+          sh(label: "Pre-compression actions", script: "mkdir -p target && cp -r ${packageVersion} target/${packageName}-${packageVersion}")
+          zip(zipFile: "${packageName}-${packageVersion}.zip", dir: "target")
+          sh(label: "Post-compression actions", script: """
+            mkdir -p ../../build/packages
+            mv ${packageName}-${packageVersion}.zip ../../build/packages/
+            rm -r target
+            """)
+        }
+      }
+    }
+  }
+}
+
+def signArtifactsWithElastic() {
+  googleStorageUpload(bucket: env.INFRA_SIGNING_BUCKET_ARTIFACTS_PATH,
+    credentialsId: env.INTERNAL_CI_JOB_GCS_CREDENTIALS,
+    pathPrefix: env.BUILD_ARTIFACTS_PATH + '/',
+    pattern: env.BUILD_ARTIFACTS_PATH + '/*.zip',
+    sharedPublicly: false,
+    showInline: true)
+  withCredentials([string(credentialsId: env.JOB_SIGNING_CREDENTIALS, variable: 'TOKEN')]) {
+    triggerRemoteJob(auth: CredentialsAuth(credentials: 'local-readonly-api-token'),
+      job: 'https://internal-ci.elastic.co/job/elastic+unified-release+master+sign-artifacts-with-gpg',
+      token: TOKEN,
+      parameters: "gcs_input_path=${env.INFRA_SIGNING_BUCKET_ARTIFACTS_PATH}",
+      useCrumbCache: true,
+      useJobInfoCache: true)
+  }
+  googleStorageDownload(bucketUri: "${env.INFRA_SIGNING_BUCKET_SIGNED_ARTIFACTS_PATH}/*",
+    credentialsId: env.INTERNAL_CI_JOB_GCS_CREDENTIALS,
+    localDirectory: env.BUILD_ARTIFACTS_PATH + '/',
+    pathPrefix: "${env.INFRA_SIGNING_BUCKET_SIGNED_ARTIFACTS_SUBFOLDER}")
+    sh(label: 'Rename .asc to .sig', script: 'for f in ' + env.BUILD_ARTIFACTS_PATH + '/*.asc; do mv "$f" "${f%.asc}.sig"; done')
+}
+
+def publishToPackageStorage() {
+  dir(env.BUILD_ARTIFACTS_PATH) {
+    withGCPEnv(secret: env.PACKAGE_STORAGE_UPLOADER_GCP_SERVICE_ACCOUNT) {
+      withCredentials([string(credentialsId: env.PACKAGE_STORAGE_UPLOADER_CREDENTIALS, variable: 'TOKEN')]) {
+        findFiles()?.findAll{ it.name.endsWith('.zip') }?.collect{ it.name }?.sort()?.each {
+          def packageZip = it
+          sh(label: 'Upload package .zip file', script: "gsutil cp ${packageZip} ${env.PACKAGE_STORAGE_INTERNAL_BUCKET_QUEUE_PUBLISHING_PATH}/")
+          sh(label: 'Upload package .sig file', script: "gsutil cp ${packageZip}.sig ${env.PACKAGE_STORAGE_INTERNAL_BUCKET_QUEUE_PUBLISHING_PATH}/")
+
+          triggerRemoteJob(auth: CredentialsAuth(credentials: 'local-readonly-api-token'),
+            job: 'https://internal-ci.elastic.co/job/package_storage/job/publishing-job-remote',
+            token: TOKEN,
+            parameters: """
+              dry_run=true
```

Comment:
@jsoriano Enabled dry run is the only piece that prevents us from syncing packages with storage v2. I was wondering if we can disable it in this PR and try to sync packages now, then merge it. It will prevent a long running CI job for the production branch. WDYT?

Reply:
LGTM, this way real packages will start to become available in the new storage, right?

Reply:
Yes, precisely. The storage will be populated with packages and it will unblock works on the Package Registry apps.

Reply:
Ok, so the job was running for 12 hours and timeouted. We may need to adjust the publishing job to handle multiple packages at once, as the bottleneck is scheduling. Otherwise, we will run it for another 20 days :)

Reply:
Maybe we can create a one-time job that publishes all the existing packages at once?

Reply:
Yeah, it would work similarly. I'm wondering which option will be more useful in the future.